rest/src/main/java/org/jboss/pnc/rest/provider/ArtifactProvider.java

```diff
@@ -89,12 +91,20 @@ public ArtifactProvider(ArtifactRepository artifactRepository, RSQLPredicateProd
                 ArtifactRest.class, fullArtifactList);
     }
 
+    /**
+     * Lookups built artifacts for the specified BuildRecord
+     * 
+     * @return Returns requested artifacts or empty collection if BuildRecord with the specified ID doesn't exists
+     */
     public CollectionInfo<ArtifactRest> getBuiltArtifactsForBuildRecord(int pageIndex, int pageSize, String sortingRsql, String query,
             int buildRecordId) {
         BuildRecord buildRecord = buildRecordRepository.queryById(buildRecordId);
-
-        return filterAndSort(pageIndex, pageSize, sortingRsql, query,
-                ArtifactRest.class, buildRecord.getBuiltArtifacts());
+        Set<Artifact> builtArtifacts = Collections.emptySet();
+        if (buildRecord != null)
+        	builtArtifacts = buildRecord.getBuiltArtifacts();
```

Comment:
my personal feelings are that a 404 in the case of missing buildRecord would say more than 204.
But I think Matej was in favor of 204 in both cases, so let's do it this way.